### DISCUSSION TITLE
What rules should I consider for ESLint and Prettier configuration?

### DISCUSSION THREAD

**Discussion by joshi-kaushal** [reactplay/react-play#277] [Ideas]
I am working on the issue https://github.com/reactplay/react-play/issues/263. I am done with configuring them both in our app. But before I test the code, I want a list of all the rules and options that will be followed for linting.

- [ESLint rules](https://eslint.org/docs/rules/)
- [Prettier options](https://prettier.io/docs/en/options.html)

Thanks.

**Comment by Sachin-chaurasiya** [comment]
Hey @joshi-kaushal, we can consider https://github.com/jsx-eslint/eslint-plugin-react for react specific rules.

- react/jsx-pascal-case
- react/jsx-sort-props
- react/jsx-boolean-value
- react/self-closing-comp
- react/jsx-curly-brace-presence
- react/prop-types
- react/display-name
- no-console

**Reply by joshi-kaushal** [reply]
I have done changes as you said, please take a look once again. (will push soon, commenting early)

`react/prop-types` and `react/display-name` are enabled in recommended configuration so I have not added that. Also, I couldn't find `no-console`  in the [list](https://github.com/jsx-eslint/eslint-plugin-react).

**Reply by joshi-kaushal** [reply]
And we have a lot to fix once we are done with this. Most of these looks are regarding indentation and [endOfLine](https://prettier.io/docs/en/options.html#end-of-line)

I haven't run `--fix` yet. 

![image](https://user-images.githubusercontent.com/53049546/172797600-0bed3408-9cf3-4b7f-babe-1341c9076b8c.png)

**Reply by Sachin-chaurasiya** [reply]
Hey @joshi-kaushal , `no-console` is in the vanilla eslint list.
You can add it like this

```
 // Console logs cannot be committed.
    "no-console": ["error"],

```

**Comment by joshi-kaushal** [comment]
@Sachin-chaurasiya 
Yes, I have installed it. I wanted to see if there are more rules that need to be specified in `eslintrc` or `prettierrc`.

[Here](https://github.com/joshi-kaushal/react-play/tree/lint) is my current progress if you want to check.

**Reply by Sachin-chaurasiya** [reply]
Hey @joshi-kaushal , `prettierrc` looks good to me.
Can we start with the above rules?
Let me know if you need any help on the same.

**Reply by atapas** [reply]
Let's roll it. I would suggest a PR to create and we can provide comment on it.

**Comment by koustov** [comment]
When you create the PR, make sure you roll out the environmental pre-requisite checklist (if any) which can be recorded in wiki
Pre-req like
1. If you need to have some specific global package
2. For different code editor, what settings you want to tweak. Something like `autoSave`
3. Is there any specific steps to follow to override code editor level configuration with local config
4. etc
